speedy-recovery-client/src/components/secured/core/SecuredMainPage.js

```diff
@@ -85,10 +85,17 @@ class SecuredMainPage extends Component {
   }
 
   updateStateAppointments = () => {
-    // TODO: Query SMART, display appointment data for current user
-    const appointments = fhirExampleApp.map((app) => fhirMapAppointment(app, this.props.fhirVersion));
+    if(this.props.user !== null){
+      FhirDataQueryingService.getUserAppointment(this.props.user.id)
+        .then(appointment => {
+          const appointments = this.handleAppointment(appointment);
+          this.setState({ appointments });
+        }).catch(errorMessage=>{this.setState([])});
```

Comment:
Is it necessary to set the state in error case? Can't you just log the error and do nothing?
Also, if you DO write it like now, you should `.catch(() => { this.setState([]) });` as you don't actually use the error message.

Reply:
What's more, I guess your intent was to set the `state.appointments` to `[]`, but to do that, you would have to `this.setState({ appointments: [] })`